### PR TITLE
First steps for generating nuget packages for ARM64/Android.

### PR DESCRIPTION
**Pull request by cydhaselton**
Adds RIDS and related conditionals/variables to the appropriate files as part of Enabling Android/ARM64 for .NET Core ([issue here](https://github.com/dotnet/core-setup/issues/1651)).

Code appears to have been refactored since the original PR to which I was referred, cc-ing @janvorli and @qmfrederik for review in case of missed items

### CODE REVIEW

**Comment by cydhaselton** [comment]
Also tagging @gkhanna79 for review because of #6056 and #6000

File: src/.nuget/Microsoft.NETCore.Native/Microsoft.NETCore.Native.props [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="utf-8"?>
 <Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <PropertyGroup>
-    <SupportedPackageOSGroups>Linux;OSX</SupportedPackageOSGroups>
+    <SupportedPackageOSGroups>Linux;Android;OSX</SupportedPackageOSGroups>
```
Comment (gkhanna79) [inline src/.nuget/Microsoft.NETCore.Native/Microsoft.NETCore.Native.props]: Is there a reason you are enabling this here?

CC @jkotas

File: src/.nuget/dir.props [hunk 2]
```diff
@@ -23,7 +23,7 @@
 
     <RuntimeOS Condition="'$(RuntimeOS)' == ''">$(OSRid)</RuntimeOS>
 
-    <SupportedPackageOSGroups Condition="'$(SupportedPackageOSGroups)' == ''">Windows_NT;OSX;Linux</SupportedPackageOSGroups>
+    <SupportedPackageOSGroups Condition="'$(SupportedPackageOSGroups)' == ''">Windows_NT;OSX;Android;Linux</SupportedPackageOSGroups>
```
Comment (gkhanna79) [inline src/.nuget/dir.props]: What is the __DistroRid for Android build?

File: src/.nuget/Microsoft.NETCore.Native/Microsoft.NETCore.Native.props (hunk 1 above)
Comment (cydhaselton) [inline src/.nuget/Microsoft.NETCore.Native/Microsoft.NETCore.Native.props]: Because Linux and OSX were enabled here. If it's not applicable I can remove; I'm basically working from PRs and extrapolating based on refactored code so some of the edits may not be needed. Feel free to let me know and I'll make changes

File: src/.nuget/dir.props (hunk 2 above)
Comment (cydhaselton) [inline src/.nuget/dir.props]: android.21-arm64 and/or android.21-arm, unless there's a difference between RID as registered in CoreFX and __DistroRid?

**Comment by cydhaselton** [comment]
@jkotas and/or @gkhanna79:  Is there a definitive list of files to which a new RID should be added and/or detected for nuget package support?

File: src/.nuget/Microsoft.NETCore.Native/Microsoft.NETCore.Native.props (hunk 1 above)
Comment (gkhanna79) [inline src/.nuget/Microsoft.NETCore.Native/Microsoft.NETCore.Native.props]: I think you can skip this file from the change.

File: src/.nuget/dir.props (hunk 2 above)
Comment (gkhanna79) [inline src/.nuget/dir.props]: Was this updated?

Comment (cydhaselton) [inline src/.nuget/dir.props]: Not yet; still working out where __DistroRid originates and/or is generated (also see question below)

Comment (cydhaselton) [inline src/.nuget/dir.props]: @gkhanna79 The test hack I added to build.sh generates __DistroRid for Android build and, with following commits, allows package build for Android.

I've added a TODO for figuring out a reliable way to generate __DistroRid for Android. Since I generated the commit changes from the `runtime.OS._filename_.props files and guesswork, a once-over to make sure I haven't missed anything would be appreciated

**Comment by cydhaselton** [comment]
@gkhanna79 The test hack I added to build.sh generates __DistroRid for Android build and, with following commits, allows package build for Android.

I've added a TODO for figuring out a reliable way to generate __DistroRid for Android. Since I generated the commit changes from the `runtime.OS._filename_.props files and guesswork, a once-over to make sure I haven't missed anything would be appreciated

**Comment by cydhaselton** [comment]
Copying @mellinoe for additional visibility into package generation

**Comment by cydhaselton** [comment]
cc-ing @qmfrederik @mellinoe and @janvorli for additional visibility into package generation

**Comment by cydhaselton** [comment]
Also copying @schellap for input

File: build.sh [hunk 3]
```diff
@@ -74,7 +74,9 @@ initTargetDistroRid()
         if [ "$__BuildOS" == "Linux" ]; then
             if [ ! -e $ROOTFS_DIR/etc/os-release ]; then
                 echo "WARNING: Can not determine runtime id for current distro."
-                export __DistroRid=""
+                #Test hack to generate __DistroRid for Android
+                #TODO Find reliable way to generate __DistroRid for Android.
+                export __DistroRid="android.21-arm64"
```
Comment (gkhanna79) [inline build.sh]: Just to be clear - we cannot commit with this change. Have you considered specifying this at command (e.g. __DistroRid=... ./build.sh) and then adding a check in this function if __DistroRid is already defined or not?

File: src/.nuget/Microsoft.NETCore.Native/Microsoft.NETCore.Native.props [hunk 4]
```diff
@@ -3,4 +3,4 @@
   <PropertyGroup>
     <SupportedPackageOSGroups>Linux;OSX</SupportedPackageOSGroups>
   </PropertyGroup>
-</Project>
\ No newline at end of file
+</Project>
```
Comment (gkhanna79) [inline src/.nuget/Microsoft.NETCore.Native/Microsoft.NETCore.Native.props]: Undo change to this file.

File: src/.nuget/Microsoft.NETCore.Runtime.CoreCLR/runtime.Android.Microsoft.NETCore.Runtime.CoreCLR.props [hunk 5]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup>
+    <_PlatformDoesNotSupportNiFiles Condition="'$(Platform)' == 'arm'">true</_PlatformDoesNotSupportNiFiles>
+    <_PlatformDoesNotSupportNiFiles Condition="'$(Platform)' == 'armel'">true</_PlatformDoesNotSupportNiFiles>
+    <_PlatformDoesNotSupportNiFiles Condition="'$(Platform)' == 'arm64'">true</_PlatformDoesNotSupportNiFiles>
+  </PropertyGroup>
+  <ItemGroup>
+    <NativeBinary Include="$(BinDir)libcoreclr.so" />
+    <NativeBinary Condition="'$(_PlatformDoesNotSupportNiFiles)' != 'true'" Include="$(BinDir)libcoreclrtraceptprovider.so" />
```
Comment (gkhanna79) [inline src/.nuget/Microsoft.NETCore.Runtime.CoreCLR/runtime.Android.Microsoft.NETCore.Runtime.CoreCLR.props]: This check is incorrect. __PlatformDoesNotSupportNiFiles_ should be used for platforms where we cannot generate NI images right now. They just happen to be same as where this binary does not build.

If you are able to build this for Android, please fix this check accordingly so that it can be included for that platform.

FYI - @hqueue @hseok-oh @jyoungyun

File: src/.nuget/dir.props [hunk 6]
```diff
@@ -149,6 +162,14 @@
       <Platform>arm64</Platform>
     </OfficialBuildRID>
   </ItemGroup>
+  <ItemGroup Condition="$(SupportedPackageOSGroups.Contains(';Android;'))">
+    <OfficialBuildRID Include="android.21-arm64">
+      <Platform>arm64</Platform>
+    </OfficialBuildRID>
+    <OfficialBuildRID Include="android.21-arm">
```
Comment (gkhanna79) [inline src/.nuget/dir.props]: Have you tested the build for Arm32 for Android? If so, your change for __DsitroRid should account for it as well.

File: build.sh (hunk 3 above)
Comment (cydhaselton) [inline build.sh]: @gkhanna79:  I did not, but that is a good idea.

File: src/.nuget/dir.props (hunk 6 above)
Comment (cydhaselton) [inline src/.nuget/dir.props]: I've only tested...and am only able to test...for arm64.  I can add arm into this change but won't be able to test it.

Comment (gkhanna79) [inline src/.nuget/dir.props]: In that case, please undo the Android for Arm change.

File: config.json [hunk 7]
```diff
@@ -327,7 +327,7 @@
     "RuntimeId": {
       "description": "Specifies the OS to build Core_Root for",
       "valueType": "property",
-      "values": [ "debian.8-x64", "fedora.23-x64", "opensuse.42.1-x64", "osx.10.12-x64", "rhel.7-x64", "ubuntu.14.04-x64", "ubuntu.16.04-x64", "ubuntu.16.10-x64" ],
+      "values": [ "debian.8-x64", "fedora.23-x64", "opensuse.42.1-x64", "osx.10.10-x64", "rhel.7-x64", "ubuntu.14.04-x64", "ubuntu.16.04-x64", "ubuntu.16.10-x64", "android.21-arm64", "android.21-arm" ],
```
Comment (jyoungyun) [inline config.json]: I think you need to undo osx version change.

File: build.sh (hunk 3 above)
Comment (jyoungyun) [inline build.sh]: IMHO, you need to save `ANDROID_PLATFORM` variable as a file when you build rootfs and then you can use it in building coreclr. :)

Comment (cydhaselton) [inline build.sh]: @jyoungyun This may be better; so far I've tried every iteration of detecting `__DistroRid` in this function and all have failed except for `! [[ -n "$__DistroRid" ]]`, which is the same as hardcoding that value.

Comment (cydhaselton) [inline build.sh]: @gkhanna79 See recent commit

File: src/.nuget/Microsoft.NETCore.Native/Microsoft.NETCore.Native.props (hunk 4 above)
Comment (cydhaselton) [inline src/.nuget/Microsoft.NETCore.Native/Microsoft.NETCore.Native.props]: @gkhanna79 Done

File: config.json (hunk 7 above)
Comment (cydhaselton) [inline config.json]: That's odd…don't remember making that change. Fixed

File: src/.nuget/dir.props (hunk 6 above)
Comment (cydhaselton) [inline src/.nuget/dir.props]: Removed

File: build.sh (hunk 3 above)
Comment (cydhaselton) [inline build.sh]: FYI, Walking back all changes to build.sh and going route @jyoungyun  recommended. Too difficult getting logic check in `initTargetDistroRid()` to work without causing CI errors

**Comment by cydhaselton** [comment]
Tagging @qmfrederik to review/ok addition to `build-android-rootfs.sh`

**Comment by cydhaselton** [comment]
General question: I'm still getting up to speed with CI builds. Is there anything I need to do with the previous commits in this PR that have failed build checks if a later commit or commits are successful?

**Comment by qmfrederik** [comment]
@cydhaselton The change to `build-android-rootfs` is fine with me.

Regarding to the CI builds, only the last commit matters. If the previous commits are no longer relevant, you can squash them, and they will disappear from the list of commits in this pull request.

Would it be easier to treat Android as a Linux distribution? That would avoid the need to add a new OS group.

**Comment by cydhaselton** [comment]
Another (probably obvious) question: is there a way for me manually to re-run a CI build for a specific commit?

**Comment by qmfrederik** [comment]
@cydhaselton You can force the CI build to re-run, but only for the last commit in your PR.

**Comment by cydhaselton** [comment]
@gkhanna79 @jyoungyun and others…any other changes/additions I should make to this PR before merge?

File: build.sh [hunk 8]
```diff
@@ -73,13 +73,18 @@ initTargetDistroRid()
...
+                else
+                    echo "WARNING: Can not determine runtime id for current distro."
+                    export __DistroRid=""
+                fi
             else
                 source $ROOTFS_DIR/etc/os-release
                 export __DistroRid="$ID.$VERSION_ID-$__BuildArch"
             fi
-        fi
+         fi
```
Comment (jyoungyun) [inline build.sh]: Remove redundant space :)

Comment (cydhaselton) [inline build.sh]: Good catch!  Fixed.

File: build.sh [hunk 9]
```diff
@@ -73,8 +73,13 @@ initTargetDistroRid()
     if [ $__CrossBuild == 1 ]; then
         if [ "$__BuildOS" == "Linux" ]; then
             if [ ! -e $ROOTFS_DIR/etc/os-release ]; then
-                echo "WARNING: Can not determine runtime id for current distro."
-                export __DistroRid=""
+                if [ -e $ROOTFS_DIR/android_platform ]; then
+                    source $ROOTFS_DIR/android_platform
+                    export __DistroRid="$RID"
```
Comment (gkhanna79) [inline build.sh]: I presume that $RID is being set by the source statement earlier,right? What is the value it gets?

File: config.json [hunk 10]
```diff
@@ -327,7 +327,7 @@
     "RuntimeId": {
       "description": "Specifies the OS to build Core_Root for",
       "valueType": "property",
-      "values": [ "debian.8-x64", "fedora.23-x64", "opensuse.42.1-x64", "osx.10.12-x64", "rhel.7-x64", "ubuntu.14.04-x64", "ubuntu.16.04-x64", "ubuntu.16.10-x64" ],
+      "values": [ "debian.8-x64", "fedora.23-x64", "opensuse.42.1-x64", "osx.10.12-x64", "rhel.7-x64", "ubuntu.14.04-x64", "ubuntu.16.04-x64", "ubuntu.16.10-x64", "android.21-arm64", "android.21-arm" ],
```
Comment (gkhanna79) [inline config.json]: Were you not planning to remove the "arm" entry?

File: build.sh (hunk 9 above)
Comment (gkhanna79) [inline build.sh]: Nevermind - saw it further below.

File: src/.nuget/Microsoft.NETCore.ILAsm/runtime.Android.Microsoft.NETCore.ILAsm.props [hunk 11]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (gkhanna79) [inline src/.nuget/Microsoft.NETCore.ILAsm/runtime.Android.Microsoft.NETCore.ILAsm.props]: Thinking a bit more about this, why do you need to add these android specific files? Isn't the Linux version sufficient?

File: src/.nuget/dir.props [hunk 12]
```diff
@@ -77,6 +78,13 @@
         <PackageRID Condition="'$(PortableBuild)' == '1'">osx-$(ArchGroup)</PackageRID>
       </PropertyGroup>
     </When>
+    <When Condition="'$(_runtimeOSFamily)' == 'android'">
+      <PropertyGroup>
+        <PackageRID>android.21-$(ArchGroup)</PackageRID>
```
Comment (gkhanna79) [inline src/.nuget/dir.props]: Should we have a check for arm64 since that is all we are supporting for now?

File: config.json (hunk 10 above)
Comment (cydhaselton) [inline config.json]: I did, but reverted when working on fixing initTargetDistroRid logic in build.sh and forgot to re-remove.
Fixed

File: src/.nuget/Microsoft.NETCore.ILAsm/runtime.Android.Microsoft.NETCore.ILAsm.props (hunk 11 above)
Comment (cydhaselton) [inline src/.nuget/Microsoft.NETCore.ILAsm/runtime.Android.Microsoft.NETCore.ILAsm.props]: I'm not sure; I'm still piecing together how nuget packages are generated. Since I don't know all of the places a) where it would be important to separate Android from Linux and b) where it wouldn't matter, I'm taking a "separate everything now, integrate with Linux after" approach.

I can remove the files and see if package build still works.

Comment (cydhaselton) [inline src/.nuget/Microsoft.NETCore.ILAsm/runtime.Android.Microsoft.NETCore.ILAsm.props]: @gkhanna79: So the build is successful, including packages, once `packageTargetOsGroup` for Android is removed from `dir.props` and conditionals added to `Microsoft.NETCore.Runtime.CoreCLR.pkgproj`

So my questions are:

1. What elements distinguish one package group from another? Or: What factors determine the creation of a new OS group?
2. Is it possible that a separate package group would be needed for Android in the future?

**Comment by cydhaselton** [comment]
@gkhanna79 @qmfrederik  and others; re-posting this question in general discussion for greater visibility. Code discussion [here](https://github.com/dotnet/coreclr/pull/10286#pullrequestreview-30691338):

So my questions are:
1. What elements distinguish one package group from another? Or: What factors determine the creation of a new OS group?
2. Is it possible that a separate package group would be needed for Android in the future?

**Comment by cydhaselton** [comment]
cc @hseok-oh and @ericstj for insight into OS Group package info

**Comment by danmosemsft** [comment]
@dotnet-bot test OSX10.12 x64 Checked Build and Test  (network)

**Comment by cydhaselton** [comment]
Also cc-ing @jkotas as per [documentation](https://github.com/dotnet/coreclr/blob/master/Documentation/project-docs/contributing-workflow.md); is there a person or persons that are responsible for the packaging subsystem that I should copy on this for feedback on the OS Group question?

**Comment by jkotas** [comment]
You have the right people copied on this already  (@gkhanna79, @ericstj).

**Comment by cydhaselton** [comment]
Thanks, @jkotas. Would it be appropriate for me to update the CODE_OWNERS.TXT with that info?

**Comment by jkotas** [comment]
Yes, it sounds good to me.

**Comment by cydhaselton** [comment]
Polite ping to @ericstj and @gkhanna79 again for input into above questions

File: src/.nuget/Microsoft.NETCore.ILAsm/runtime.Android.Microsoft.NETCore.ILAsm.props (hunk 11 above)
Comment (gkhanna79) [inline src/.nuget/Microsoft.NETCore.ILAsm/runtime.Android.Microsoft.NETCore.ILAsm.props]: Given that Android is sufficiently distinct, but still Linux based, I am fine if a new OS group for it is used while leveraging the existing Linux definitions of the components.

CC @chcosta

File: src/.nuget/Microsoft.NETCore.Runtime.CoreCLR/runtime.Android.Microsoft.NETCore.Runtime.CoreCLR.props [hunk 13]
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup>
+    <_PlatformDoesNotSupportNiFiles Condition="'$(Platform)' == 'arm'">true</_PlatformDoesNotSupportNiFiles>
+    <_PlatformDoesNotSupportNiFiles Condition="'$(Platform)' == 'armel'">true</_PlatformDoesNotSupportNiFiles>
+    <_PlatformDoesNotUseNiFiles Condition="'$(Platform)' == 'arm64'">true</_PlatformDoesNotUseNiFiles>
+  </PropertyGroup>
+  <ItemGroup>
+    <NativeBinary Include="$(BinDir)libcoreclr.so" />
+    <NativeBinary Condition="'$(_PlatformDoesNotSupportNiFiles)' != 'true'" Include="$(BinDir)libcoreclrtraceptprovider.so" />
```
Comment (gkhanna79) [inline src/.nuget/Microsoft.NETCore.Runtime.CoreCLR/runtime.Android.Microsoft.NETCore.Runtime.CoreCLR.props]: This just got recently changed to be supported. Thus, keeping it unified with the Linux definition will help streamline where it needs to be enabled vs not.

Comment (cydhaselton) [inline src/.nuget/Microsoft.NETCore.Runtime.CoreCLR/runtime.Android.Microsoft.NETCore.Runtime.CoreCLR.props]: Should I revert this change then or keep it as is with the separate `PlatformDoesNotUseNiFiles` condition?

File: src/.nuget/Microsoft.NETCore.ILAsm/runtime.Android.Microsoft.NETCore.ILAsm.props (hunk 11 above)
Comment (cydhaselton) [inline src/.nuget/Microsoft.NETCore.ILAsm/runtime.Android.Microsoft.NETCore.ILAsm.props]: Unfortunately, unless I'm missing something, `packageTargetOSGroup` means adding Android-specific runtime-prefixed files (`runtime.Android...`).

File: src/.nuget/Microsoft.NETCore.Runtime.CoreCLR/runtime.Android.Microsoft.NETCore.Runtime.CoreCLR.props (hunk 13 above)
Comment (cydhaselton) [inline src/.nuget/Microsoft.NETCore.Runtime.CoreCLR/runtime.Android.Microsoft.NETCore.Runtime.CoreCLR.props]: cc @gkhanna79

File: src/.nuget/Microsoft.NETCore.ILAsm/runtime.Android.Microsoft.NETCore.ILAsm.props (hunk 11 above)
Comment (cydhaselton) [inline src/.nuget/Microsoft.NETCore.ILAsm/runtime.Android.Microsoft.NETCore.ILAsm.props]: cc @gkhanna79

Comment (chcosta) [inline src/.nuget/Microsoft.NETCore.ILAsm/runtime.Android.Microsoft.NETCore.ILAsm.props]: I think it's reasonable for Android to use the Linux `_packageTargetOSGroup` to import the runtime package imports

File: src/.nuget/Microsoft.NETCore.Runtime.CoreCLR/runtime.Android.Microsoft.NETCore.Runtime.CoreCLR.props (hunk 13 above)
Comment (danmosemsft) [inline src/.nuget/Microsoft.NETCore.Runtime.CoreCLR/runtime.Android.Microsoft.NETCore.Runtime.CoreCLR.props]: @gkhanna79 for q above.

File: src/.nuget/Microsoft.NETCore.ILAsm/runtime.Android.Microsoft.NETCore.ILAsm.props (hunk 11 above)
Comment (cydhaselton) [inline src/.nuget/Microsoft.NETCore.ILAsm/runtime.Android.Microsoft.NETCore.ILAsm.props]: @chcosta and @gkhanna79: Still not sure how to achieve what you are suggesting; if I create a `packageTargetOSGroup` for Android without adding the `runtime.Android` files package build fails.  Is there another OS that does this that I could reference?

File: src/.nuget/dir.props [hunk 14]
```diff
@@ -34,6 +34,7 @@
   <!-- derive an OS Group based on the OS Family -->
   <PropertyGroup>
     <_derivedPackageTargetOSGroup Condition="'$(_derivedPackageTargetOSGroup)' == '' and '$(_runtimeOSFamily)' == 'osx'">OSX</_derivedPackageTargetOSGroup>
+    <_derivedPackageTargetOSGroup Condition="'$(_derivedPackageTargetOSGroup)' == '' and '$(_runtimeOSFamily)' == 'android'">Android</_derivedPackageTargetOSGroup>
```
Comment (chcosta) [inline src/.nuget/dir.props]: @cydhaselton, `_packageTargetOSGroup` is the property which handles the runtime package import.  If you deleted this line, then I'd expect it to default to importing "Linux" runtime package projects.  That's probably not quite what we want though because it would then be ignoring the "Android" piece of the `SupportedPackageOSGroups`.  Instead, I would leave this code as-is, but add a conditional property definition here https://github.com/dotnet/coreclr/blob/8e986affba47efc1ce85ab0bb9428a6c28dae279/src/.nuget/dir.props#L47, which is something like

```
<_packageTargetOSGroup Condition="'$(_derivedPackageTargetOSGroup)' == 'Android'">Linux</_packageTargetOSGroup>
```

and maybe add a comment explaining that Android is using the same runtime pkgproj's as Linux.
```

Comment (cydhaselton) [inline src/.nuget/dir.props]: @chcosta FYI, Implementing the above causes the package build for `Microsoft.NETCore.Runtime.CoreCLR` to fail. Specifically the build ignores the conditional I've copied from the `runtime.Android` file to the `runtime.Linux`file  to exclude `System.Private.Corelib.Ni.dll` from the package. If I use the existing condition for Ni  it does work…but since `Ni` files can be generated post-build on Android it shouldn't be used.

Comment (cydhaselton) [inline src/.nuget/dir.props]: Polite ping to @gkhanna79 and @ericstj  for input into above question.

Comment (gkhanna79) [inline src/.nuget/dir.props]: @cydhaselton SPC.ni.dll generation has changed in the last week - can you please rebase and see if you are still running into this issue?

Comment (cydhaselton) [inline src/.nuget/dir.props]: @gkhanna79 I am not; rebase resolved the issue. See updated code

**Comment by cydhaselton** [comment]
@gkhanna79 The above looks messy; did you mean "merge" instead of "rebase?

**Comment by gkhanna79** [comment]
What is looking messy?

**Comment by cydhaselton** [comment]
> What is looking messy?
@gkhanna79 Ignore…see edited message. Basically didn't do `git push -f` after rebase.

**Comment by cydhaselton** [comment]
@dotnet-bot test Windows_NT x64 Debug Build and Test

**Comment by cydhaselton** [comment]
@gkhanna79, @ericstj and others…is there anything else that needs to be done for this PR?

File: config.json [hunk 15]
```diff
@@ -345,7 +345,7 @@
     "RuntimeId": {
       "description": "Specifies the OS to build Core_Root for",
       "valueType": "property",
-      "values": [ "debian.8-x64", "fedora.24-x64", "fedora.25-x64", "opensuse.42.1-x64", "osx.10.12-x64", "rhel.7-x64", "ubuntu.14.04-x64", "ubuntu.16.04-x64", "ubuntu.16.10-x64" ],
+      "values": [ "debian.8-x64", "fedora.23-x64", "fedora.25-x64", "opensuse.42.1-x64", "osx.10.12-x64", "rhel.7-x64", "ubuntu.14.04-x64", "ubuntu.16.04-x64", "ubuntu.16.10-x64", "android.21-arm64" ],
```
Comment (ericstj) [inline config.json]: This change to fedora appears to be a bad merge.

File: src/.nuget/dir.props [hunk 16]
```diff
@@ -102,6 +111,13 @@
         <SymbolFileExtension>.dwarf</SymbolFileExtension>
       </PropertyGroup>
     </When>
+    <When Condition="'$(_runtimeOSFamily)' == 'android'">
+      <PropertyGroup>
+        <LibraryFileExtension>.so</LibraryFileExtension>
+<!--symbols included in .so, like Linux, but can be generated externally and if so, uses .debug ext-->
```
Comment (ericstj) [inline src/.nuget/dir.props]: nit: indentation

File: config.json (hunk 15 above)
Comment (cydhaselton) [inline config.json]: Crap. Fixed

File: src/.nuget/dir.props (hunk 16 above)
Comment (cydhaselton) [inline src/.nuget/dir.props]: Fixed

**Comment by ericstj** [comment]
I've reviewed and commented on stuff that stuck out to me.  I don't see anything else concerning.  Most of these changes are around how coreclr decides to factor its package source which I don't have an opinion on so long as it produces the correct results.  I'd like for @gkhanna79 or @chcosta to give explicit sign off on that.

**Comment by cydhaselton** [comment]
Thanks Eric.
@gkhanna79 and/or @chcosta let me know if there's anything you see that needs to be fixed.

**Comment by cydhaselton** [comment]
Polite re-ping to @gkhanna79 and @chcosta … is there anything I need to change in this PR before merge?

File: cross/build-android-rootfs.sh [hunk 17]
```diff
@@ -136,7 +137,11 @@ dpkg -x $__Android_Cross_Dir/deb/libunwind_1.2.20170304_$__AndroidArch.deb $__An
 
 cp -R $__Android_Cross_Dir/tmp/$__AndroidArch/data/data/com.termux/files/usr/* $__ToolchainDir/sysroot/usr/
 
+# Generate platform file for build.sh script to assign to __DistroRid
+# TODO: replace with logic that generates RID based on $__ApiLevel and $__BuildArch
+echo "Generating platform file..."
 
+echo "RID=android.21-arm64" > $__ToolchainDir/sysroot/android_platform
```
Comment (chcosta) [inline cross/build-android-rootfs.sh]: Any reason not to address this TODO and use the ApiLevel / BuildArch here?

File: src/.nuget/dir.props [hunk 18]
```diff
@@ -163,6 +179,11 @@
       <Platform>arm64</Platform>
     </OfficialBuildRID>
   </ItemGroup>
+  <ItemGroup Condition="$(SupportedPackageOSGroups.Contains(';Android;'))">
+    <OfficialBuildRID Include="android.21-arm64">
```
Comment (chcosta) [inline src/.nuget/dir.props]: Do you want to include the Portable rid?

**Comment by chcosta** [comment]
Two additional comments, and then LGTM

File: cross/build-android-rootfs.sh (hunk 17 above)
Comment (cydhaselton) [inline cross/build-android-rootfs.sh]: @chcosta The TODO was addressed in previous commit by adding logic to `build-android-rootfs.sh`; I accidentally left the comment. Will remove comment and echo commands.

File: src/.nuget/dir.props (hunk 18 above)
Comment (cydhaselton) [inline src/.nuget/dir.props]: @chcosta At this line? I don't think so…a quick check of master shows that the other supported OSs don;t include it at this point. Should it be?

@gkhanna79 Anything you see that needs changing?

Comment (gkhanna79) [inline src/.nuget/dir.props]: OfficialBuildRID is only used for platform targets that the official .NET Core build is going to support and we should add it only when we enable building for it. Otherwise, this is not required.

@cydhaselton Can you please undo this change for now?

Comment (cydhaselton) [inline src/.nuget/dir.props]: Out of town but will push change when I return.

Comment (cydhaselton) [inline src/.nuget/dir.props]: @gkhanna79 Removed

**Comment by cydhaselton** [comment]
@dotnet-bot test Tizen armel Cross Debug Build and Windows_NT x64 Debug Build and Test

**Comment by cydhaselton** [comment]
@dotnet-bot test Windows_NT x64 Debug Build and Test

**Comment by cydhaselton** [comment]
@gkhanna79 and @chcosta are there any other changes needed for this PR?

**Comment by cydhaselton** [comment]
FYI will be keeping this branch around until I have time to write up documentation.
